src/Illuminate/Support/ReadableHigherOrderCollectionProxy.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Illuminate\Support;
+
+class ReadableHigherOrderCollectionProxy
+{
+    /**
+     * The collection being operated on.
+     *
+     * @var \Illuminate\Support\Collection
+     */
+    protected $collection;
+
+    /**
+     * The method being proxied.
+     *
+     * @var string
+     */
+    protected $method;
+
+    /**
+     * Create a new proxy instance.
+     *
+     * @param  \Illuminate\Support\Collection  $collection
+     * @param  string  $method
+     * @return void
+     */
+    public function __construct(Collection $collection, $method)
+    {
+        $this->method = $method;
+        $this->collection = $collection;
+    }
+
+    public static function perform(Collection $collection, $method)
+    {
+        $action = 'filter';
+
+        foreach ($collection->getProxies() as $proxy) {
+            if (starts_with($method, $proxy)) {
+                $action = $proxy;
+            }
+        }
+
+        $method = str_replace(Str::cartesian($collection->getProxies(), ['On', 'By', 'Then']), '', $method);
```

Comment:
It could be problematic if future "filler" words were to be added, other than just `on` `by` and `then`. But, I didn't want to add another property requiring a getter.

It would be nice if it worked the same way as proxies though, since that allows userland code to add their own proxy, if it was relevant to add your own "filler", then it'd be a nice solution.

Thoughts?